devel/websocket-api.rst

```diff
@@ -0,0 +1,152 @@
+    ...
+    UseWebSocket = 1
+
+This will essentially add the following snippet to be run on the Zeek manager:
+
+.. code-block:: zeek
+   :caption: websocket.zeek
+
+   event zeek_init()
+        {
+        Cluster::listen_websocket([
+            $listen_addr=127.0.0.1,
+            $listen_port=27759/tcp,
+        ]);
+        }
+
+
+To verify if the WebSocket API is functional on your deployment, it is possible
+to use `websocat <https://github.com/vi/websocat>`_ for a quick check.
+
+.. code-block:: shell
+
+   $ echo '[]' | websocat ws://127.0.0.1:27759/v1/messages/json
+   {"type":"ack","endpoint":"3eece35d-9f94-568d-861c-6a16c433e090-websocket-2","version":"8.0.0-dev.684"}
+
+Zeek's ``cluster.log`` file will also have an entry for the WebSocket client connection.
+The empty array in the command is the array holding the client's subscriptions.
+
+Version 1
+=========
+
+The currently implemented protocol is accessible at ``v1/messages/json``. The
+`data representation <https://docs.zeek.org/projects/broker/en/current/web-socket.html#data-representation>`_
+is documented in detail within the Broker project. Note that this format is a
+direct translation of Broker's binary format into JSON, resulting in a fairly
+tight coupling between WebSocket clients and the Zeek scripts. Most prominently
+is the representation of record values as vectors instead of objects, making
+the protocol sensitive against reordering or introduction of optional fields
+into records.
+
+
+Handshake and Acknowledgement
+-----------------------------
+
+The first message after a WebSocket connection has been established originates
+from the client. This message is a JSON array of strings that represent the
+topics the WebSocket client wishes to subscribe to.
+
+Zeek replies with an acknowledgement message that's a JSON object or an error.
+
+Events
+------
+
+After the acknowledgement, WebSocket clients receive all messages arriving on
+the topics they subscribed to.
+
+.. code-block:: shell
+
+   $ websocat ws://127.0.0.1:27759/v1/messages/json
+   ["zeek.test"]
+   {"type":"ack","endpoint":"d955d990-ad8a-5ed4-8bc5-bee252d4a2e6-websocket-0","version":"8.0.0-dev.684"}
+   {"type":"data-message","topic":"zeek.test","@data-type":"vector","data":[{"@data-type":"count","data":1},{"@data-type":"count","data":1},{"@data-type":"vector","data":[{"@data-type":"string","data":"hello"},{"@data-type":"vector","data":[{"@data-type":"count","data":3}]},{"@data-type":"vector","data":[]}]}]}
+
+The received messages, again, are encoded in Broker's JSON format. Above ``data-message``
+represents an event received on topic ``zeek.test``. The event's name is ``hello``.
+This event has a single argument of type :zeek:type:`count`. In the example above
+its value is ``3``.
+
+To send events, WebSocket clients similarly encode their event representation
+to Broker's JSON format and send them as `text data frames <https://datatracker.ietf.org/doc/html/rfc6455#section-5.6>`_.
```

Comment:
Probably worth adding:


```suggestion
To send events, WebSocket clients similarly encode their event representation
to Broker's JSON format and send them as `text data frames <https://datatracker.ietf.org/doc/html/rfc6455#section-5.6>`_. Clients can only publish to topics they are subscribed to.
```

Reply:
I think I started believing that when writing my client lib, but I do not see it anymore.